docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,202 @@
+
+(Requirement 7) In order to account for a future possibility of adding additional policy options, the value of this annotation should be an enumeration. Considering the name already references the existing CloudFormation `DeletionPolicy` field, the values should similarly follow those values:
+
+* `Delete` - deletes the AWS resource alongside the CR
+* `Retain` - retains the AWS resource, but deletes the CR
+
+The default value for this annotation will be `Delete` as it is in-line with the existing ACK functionality of deleting the underlying AWS resource.
+
+#### Examples
+
+Definition at the controller - all resources reconciled by this controller will use the “retain” policy (unless overridden by an annotation):
+
+```yaml
+containers:
+    - command:
+    - ./bin/controller
+    args:
+    - --aws-account-id
+    - "$(AWS_ACCOUNT_ID)"
+    - --aws-region
+    - "$(AWS_REGION)"
+    - --enable-development-logging
+    - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
+    - --log-level
+    - "$(ACK_LOG_LEVEL)"
+    - --resource-tags
+    - "$(ACK_RESOURCE_TAGS)"
++  - --deletion-policy
++  - "retain"
+```
+
+Definition at the namespace (protecting all ACK resources across any service):
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    services.k8s.aws/deletion-policy: retain
+  name: my-retained-namespace
+```
+
+Definition at the namespace (protecting only S3 ACK resources):
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    s3.services.k8s.aws/deletion-policy: retain
+  name: my-retained-namespace
+```
+
+Definition at the resource:
+
+```yaml
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  name: test-bucket
+  annotations:
+    s3.services.k8s.aws/deletion-policy: retain
+spec:
+  name: retained-bucket
+```
+
+### Delete Protection
+
+Proposed annotation names:
+
+* `no-delete`
+* `block-delete` / `prevent-delete` / `lock-delete`
+* `delete-lock`
+
+Acceptable values for this annotation will only need to be boolean `true/false` . The annotation set to `true` will indicate to the controller that it, or any ACK resources within its scope, should not be deleted. The default value for this annotation (when the annotation is omitted) will be `false` - the resource can be deleted.
+
+When a user attempts to delete a resource, a validating webhook will read through each of the annotations on the resource and the namespace. If the premature deletion annotation is defined on either of those, if the one with the higher precedence has the value set to `true`, the validating webhook will reject the delete request with a corresponding error message. 
+
+#### Examples
+
+Definition at the namespace (protecting all ACK resources across any service):
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    services.k8s.aws/prevent-delete: true
```

Comment:
Clarifying Question:

Do you know what the namespace deletion experience is like when some custom resources fail to delete in the namespace?